### ISSUE TITLE
bottom.QuantumCircuit.tensor(top. inplace=True) doubles ancillas on modified bottom

### ISSUE THREAD

**Issue by learn2flask** [Qiskit/qiskit-terra#7445]
### Environment

- **Qiskit Terra version**:  0.19.1
- **Python version**: 3.8.5
- **Operating system**: Win11x64

### What is happening?

Given QuantumCircuits "bottom" and "top",
if "bottom" contains ancillas AND if performing bottom.tensor(top, inplace=True)
then the modified "bottom" QuantumCircuit will have the ancilla entries doubled (with same name, so they are not visible in draw) in bottom.ancillas
This does NOT happen if inplace=False

### How can we reproduce the issue?
```python
bell = QuantumCircuit(2)
bell.h(0)
bell.cx(0, 1)

c = QuantumCircuit(QuantumRegister(2), ClassicalRegister(2), AncillaRegister(2))
print(c.ancillas)

# output: 
# [AncillaQubit(AncillaRegister(2, 'a8'), 0),
# AncillaQubit(AncillaRegister(2, 'a8'), 1)]

c.tensor(bell, inplace=True)
print(c.ancillas)

# output: 
# [AncillaQubit(AncillaRegister(2, 'a8'), 0),
# AncillaQubit(AncillaRegister(2, 'a8'), 1),
# AncillaQubit(AncillaRegister(2, 'a8'), 0),
# AncillaQubit(AncillaRegister(2, 'a8'), 1)]
```

### What should happen?

Only the relevant ancillas should stay.

### Any suggestions?

 a related question:
QuantumCircuit.qubits shows all qubits, including ancilla. Is that by design?

**Comment by yjt98765** [comment]
@learn2flask I have reproduced the error, but I disagree that it does not happen for `inplace=False`. For that case, I use the following code:
```
bell = QuantumCircuit(2)
bell.h(0)
bell.cx(0, 1)

c = QuantumCircuit(QuantumRegister(2), ClassicalRegister(2), AncillaRegister(2))
print(c.ancillas)
d = c.tensor(bell, inplace=False)
print(d.ancillas)
```
Note that when `inplace=False`, the function returns a new circuit instead of modifying `c`. Therefore, although `c` seems fine (because it is not modified), the result is still wrong.

The cause of this issue is that when a new circuit is constructed within `tensor`, the `__init__` method calls `add_register`, which fails to examine the existence of `AncillaQubit` before adding. I have fixed it in #7450.
